src/qtt/pgeometry.py

```diff
@@ -270,40 +270,36 @@ def decorate(func):
 
 
 @static_var("time", {'default': 0})
-def tprint(string, dt=1, output=False, tag='default'):
+def tprint(string: str, dt: float = 1, output: bool = False, tag: str = 'default') -> Union[bool, None]:
     """ Print progress of a loop every dt seconds
 
     Args:
-        string (str): text to print
-        dt (float): delta time in seconds
-        output (bool): if True return whether output was printed or not
+        string: text to print
+        dt: delta time in seconds
+        output: if True return whether output was printed or not
         tag (str): optional tag for time
     Returns:
-        output (bool)
+        Output (bool) or None
 
     """
-    if (time.time() - tprint.time.get(tag, 0)) > dt:
+    if (time.perf_counter() - tprint.time.get(tag, 0)) > dt:
         print(string)
-        tprint.time[tag] = time.time()
+        tprint.time[tag] = time.perf_counter()
         if output:
             return True
         else:
-            return
+            return None
```

Comment:
Optionally this can be changed to:

```
    if (time.perf_counter() - tprint.time.get(tag, 0)) > dt:
        print(string)
        tprint.time[tag] = time.perf_counter()
        if output:
            return True
    elif output:
        return False
    return None
```